public/app/features/scenes/components/NestedScene.tsx

```diff
@@ -0,0 +1,108 @@
+import { css } from '@emotion/css';
+import React from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Stack } from '@grafana/experimental';
+import { Button, ToolbarButton, useStyles2 } from '@grafana/ui';
+
+import { SceneObjectBase } from '../core/SceneObjectBase';
+import {
+  SceneObject,
+  SceneObjectState,
+  SceneLayoutState,
+  SceneComponentProps,
+  isSceneLayoutObject,
+} from '../core/types';
+
+interface NestedSceneState extends SceneObjectState {
+  title: string;
+  isCollapsed?: boolean;
+  canCollapse?: boolean;
+  canRemove?: boolean;
+  layout: SceneObject<SceneLayoutState>;
+  actions?: SceneObject[];
+}
+
+export class NestedScene extends SceneObjectBase<NestedSceneState> {
+  static Component = NestedSceneRenderer;
+
+  onToggle = () => {
+    this.setState({
+      isCollapsed: !this.state.isCollapsed,
+      size: {
+        ...this.state.size,
+        ySizing: this.state.isCollapsed ? 'fill' : 'content',
+      },
+    });
```

Comment:
A thing to consider- shouldn't the toggle cause the query runner to refresh the results? Can imagine a row collapsed for a few minutes that would show outdated results when expanded.

Reply:
yes, but that is something the query runner inside that collapsed scene check when it's re-activated